src/auth/dto/create-user.dto.ts

```diff
@@ -0,0 +1,49 @@
+import {
+  IsString,
+  MinLength,
+  Matches,
+  IsNotEmpty,
+  IsEmail,
+  Length,
+} from 'class-validator';
+import { ApiProperty } from '@nestjs/swagger';
+export class CreateUserDto {
+  @IsString()
+  @ApiProperty({
+    example: 'username',
+  })
+  @IsNotEmpty({ message: 'First name is required' })
+  name: string;
+
+  @Length(8, 18, {
+    message: 'Phone number should be between 8 and 18 digits',
+  })
```

Comment:
PullProofAI

The `phone` field lacks a validation decorator for ensuring it meets the length requirement.



<!-- This is an auto-generated comment by PullProofAI -->